packages/babel-plugin-transform-decorators-2/test/fixtures/examples/overrider/exec.js

```diff
@@ -0,0 +1,17 @@
+function overrider(fn) {
+  return function(descriptor) {
+    descriptor.value = fn;
+    return {descriptor, extras: [], finishers: []};
+  }
+}
+
+class Foo {
+  @overrider(() => 3) method() {
```

Comment:
What is `this` here? Is it the constructor, the outer `this`, or the newly created instance?

Could be worth having a test for it.